tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -94,14 +114,257 @@ public int Serialize(ref byte[] bytes, int offset, Span value, IFormatterResolve
                 tagProcessors = tracer.TracerManager?.TagProcessors;
             }
 
-            offset += value.Tags.SerializeTo(ref bytes, offset, value, tagProcessors);
+            offset += SerializeTags(ref bytes, offset, value, value.Tags, tagProcessors);
 
             return offset - originalOffset;
         }
 
+        private int SerializeTags(ref byte[] bytes, int offset, Span span, ITags tags, ITagProcessor[] tagProcessors)
+        {
+            int originalOffset = offset;
+
+            offset += WriteTags(ref bytes, offset, span, tags, tagProcessors);
+            offset += WriteMetrics(ref bytes, offset, span, tags, tagProcessors);
+
+            return offset - originalOffset;
+        }
+
+        // TAGS
+
+        private int WriteTags(ref byte[] bytes, int offset, Span span, ITags tags, ITagProcessor[] tagProcessors)
+        {
+            int originalOffset = offset;
+
+            // Start of "meta" dictionary. Do not add any string tags before this line.
+            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _metaBytes);
+
+            int count = 0;
+
+            // We don't know the final count yet, write a fixed-size header and note the offset
+            var countOffset = offset;
+            offset += MessagePackBinary.WriteMapHeaderForceMap32Block(ref bytes, offset, 0);
+
+            // add "language=dotnet" tag to all spans, except those that
+            // represents a downstream service or external dependency
+            if (tags is not InstrumentationTags { SpanKind: SpanKinds.Client or SpanKinds.Producer })
+            {
+                count++;
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _languageNameBytes);
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _languageValueBytes);
+            }
```

Comment:
This is changing to "add to all spans without exceptions." I wonder if we should just do it now while you're in here, or wait for a separate PR.

This suggestion removes the condition:
```suggestion
            // add "language=dotnet" tag to all spans,
            count++;
            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _languageNameBytes);
            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, 
```

Reply:
I prefer to do all behaviour changes in a different PR. The scope for this PR was only separation of concerns.